src/telemetry/telemetry.ts

```diff
@@ -11,6 +11,31 @@ type EventResult = {
     error?: Error;
 };
 
+/**
+ * Checks if telemetry is currently enabled
+ * This is a method rather than a constant to capture runtime config changes
+ *
+ * Follows the Console Do Not Track standard (https://consoledonottrack.com/)
+ * by respecting the DO_NOT_TRACK environment variable
+ */
+export function isTelemetryEnabled(): boolean {
+    // Check if telemetry is explicitly disabled in config
+    if (config.telemetry === "disabled") {
+        return false;
+    }
+
+    const doNotTrack = process.env.DO_NOT_TRACK;
```

Comment:
```suggestion
    const doNotTrack = "DO_NOT_TRACK" in process.env;
```

we normally check if env var is set and don't care about the value.

Reply:
thx, updated